src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const copyTotal = total;
+
+      total = 0;
+
+      return copyTotal;
+    }
+
+    total += number;
```

Comment:
Consider adding a type check for `number` to ensure that only numeric values are added to the total. This can prevent potential bugs if the function is used with non-numeric arguments.